docs/client_reference.rst

```diff
@@ -2315,3 +2350,17 @@ Hierarchy of exceptions
     * :exc:`WSServerHandshakeError`
 
   * :exc:`InvalidURL`
+
+    * :exc:`InvalidUrlClientError`
+
+      * :exc:`InvalidUrlRedirectClientError`
+
+    * :exc:`NonHttpUrlClientError`
+
+      * :exc:`NonHttpUrlRedirectClientError`
+
+  * :exc:`RedirectClientError`
```

Comment:
So Sphinx supports mermaid, just like GitHub does. Can we paste the diagram from https://github.com/aio-libs/aiohttp/pull/6722#discussion_r1477103562 here? It might require enabling an extra Sphinx plugin in `docs/conf.py`.

Reply:
You want me to paste this diagram just in this file or instead of some lines?

Reply:
To be honest I don't know what You expect here - I can copy this graph
(

---
title: Exception hierarchy for invalid and non-HTTP URLs, direct or redirect
---

graph BT;

    base-client-error{{"
        ClientError
    "}};

    base-value-error{{"
        ValueError
    "}};

    base-invalid-url-error{{"
        InvalidURL
    "}};

    redirect-client-error["
        RedirectClientError
    "];

    invalid-url-client-error["
        InvalidUrlClientError
    "];

    non-http-url-client-error["
        NonHttpUrlClientError
    "];

    invalid-url-redirect-client-error[["
        InvalidUrlRedirectClientError
    "]];

    non-http-url-redirect-client-error[["
        NonHttpUrlRedirectClientError
    "]];

    subgraph "Base classes for the end-users to rely on";
        invalid-url-client-error;
        redirect-client-error;
        non-http-url-client-error;
    end;

    base-invalid-url-error --> base-value-error;
    base-invalid-url-error --> base-client-error;

    invalid-url-client-error --> base-invalid-url-error;
    redirect-client-error --> base-client-error;
    non-http-url-client-error --> base-client-error;

    non-http-url-redirect-client-error --> redirect-client-error;
    non-http-url-redirect-client-error --> non-http-url-client-error;

    invalid-url-redirect-client-error --> invalid-url-client-error;
    invalid-url-redirect-client-error --> redirect-client-error;

)
^ like this but pasting it in this doc does not look well :thinking:

Reply:
Resolvig for now as it's last unresolved conversation - You can reopen it later or just commit changes directly

Reply:
@setla did you enable the mermaid extension in Sphinx, though?